default/python/AI/InvasionAI.py

```diff
@@ -703,3 +702,20 @@ def assign_invasion_fleets_to_invade():
     for fid in FleetUtilsAI.extract_fleet_ids_without_mission_types(all_invasion_fleet_ids):
         this_mission = aistate.get_fleet_mission(fid)
         this_mission.check_mergers(context="Post-send consolidation of unassigned troops")
+
+
+def _get_base_troopers(sys_id: SystemId) -> float:
+    """Get troopers already in system, including the bases in the build queue."""
+    aistate = get_aistate()
+    this_sys_status = aistate.systemStatus.get(sys_id, {})
+    tally = 0
+    for _fid in this_sys_status.get("myfleets", []):
+        tally += FleetUtilsAI.count_troops_in_fleet(_fid)
+    production_queue = fo.getEmpire().productionQueue
+    for element in production_queue:
+        if element.buildType == EmpireProductionTypes.BT_SHIP:
```

Comment:
Let's extract code inside the cycle to a separate function. 

```suggestion
    for element in production_queue:
        tally += _get_queued_item_tally(element)
```